api/eds.proto

```diff
@@ -75,38 +75,53 @@ message LbEndpoint {
   Metadata metadata = 3;
 
   // The optional load balancing weight of the upstream host, in the range 1 -
-  // 100. Envoy uses the load balancing weight in some of the built in load
+  // 128. Envoy uses the load balancing weight in some of the built in load
   // balancers. The load balancing weight for an endpoint is divided by the sum
   // of the weights of all endpoints in the endpoint's locality to produce a
   // percentage of traffic for the endpoint. This percentage is then further
   // weighted by the endpoint's locality's load balancing weight from
   // LocalityLbEndpoints. If unspecified, each host is presumed to have equal
   // weight in a locality.
+  // The limit of 128 is somewhat arbitrary, but is applied due to performance
+  // concerns with the current implmenetation and can be removed when
+  // https://github.com/envoyproxy/envoy/issues/1285 is fixed.
   google.protobuf.UInt32Value load_balancing_weight = 4;
```

Comment:
Optionally add validate.proto constraints here (we can clean this up during the doc pass if you rather just do the weighted implementation work.

Reply:
Yeah, I think I've spent enough time shaving this particular yak :-)